### ISSUE TITLE
Transient fail in `t/100-mdif-rewrite.t`

### ISSUE THREAD

**Issue by mohawk2** [KJ7LNW/perl-PDL-IO-Touchstone#6]
From a run just now of PDL downstream CI:

```
#8 102.9 /tmp/kS1p3Vejo0:6: expected 1 fields of port data but found 2: 1 != 2 at /root/.cpanm/work/1716809048.6/PDL-IO-Touchstone-1.014/blib/lib/PDL/IO/MDIF.pm line 70.
#8 102.9 # Looks like your test exited with 25 before it could output anything.
#8 102.9 t/100-mdif-rewrite.t ........ 
```

On re-running, it passed. This should probably be investigated. The test file does a read of test data, roundtrip-writes it out, then reads it, without any `eval`. It would probably be better to read the data, compare that to a fixed set of data, then write out the fixed data and compare the file contents.

**Comment by KJ7LNW** [comment]
I saw an issue like this in Perl testers, but I was not able to understand the cause. Since most testers are succeeding I assumed it was a problem with the system doing the test, but maybe it is more than that.

- Are you able to reliably reproduce this? If so, can you provide me with a procedure to do so?
- What version Perl?

**Comment by mohawk2** [comment]
It just runs the tests. Can you try automatically running the tests e.g. 1000 times on your system (with latest PDL etc) and see if you get any failures?

**Comment by KJ7LNW** [comment]
here it goes... we'll see if it fails:

```
perl-5.40.0]$ while perl -Ilib t/100-mdif-rewrite.t &> mdif-test.log; do true; done
```

**Comment by KJ7LNW** [comment]
btw, was this an unusual perl build like long-double or 64bit-ints?

**Comment by KJ7LNW** [comment]
it did fail after 300+ tries, so I have something to troubleshoot with.

**Comment by KJ7LNW** [comment]
not sure yet, but I just got this:

```
PDL: PDL::Ops::minus(a,b,c): Parameter 'b':
  Mismatched implicit broadcast dimension 0: size 401 vs. 37
There are 3 PDLs in the expression; 1 broadcast dim.
   PDL IN EXPR.    BROADCAST DIMS
   #  0 (normal):      401
   #  1 (normal):       37
   #  2 (null)

.. at ops.pd line 136.
	PDL::Ops::__ANON__(PDL=SCALAR(0x561a06fa41d8), PDL=SCALAR(0x561a06fe0698), "") called at t/100-mdif-rewrite.t line 52
	main::verify_one(PDL=SCALAR(0x561a06fa41d8), PDL=SCALAR(0x561a06fe0698), "freqs") called at t/100-mdif-rewrite.t line 36
# Looks like your test exited with 255 just after 144.
```

**Comment by KJ7LNW** [comment]
oh, I ran out of disk space... so thats not the bug.

**Comment by mohawk2** [comment]
> btw, was this an unusual perl build like long-double or 64bit-ints?

No, just the normal PDL CI setup. It uses the GitHub Action `setup-perl`.

**Comment by KJ7LNW** [comment]
If the temp filename contains `/s(\d+)p/` then it thinks $1 is the port count.  This is _supposed_ to match the file extension, but needs to be more specific.

For example, `/tmp/S6prNa_ZsE` just randomly contains "S6p" and triggers "expected 6 fields of port data but found 2".

Also, this is really a carp not a croak from a user's perspective, because it is just that the file extension doesn't match what was in the file.

**Comment by KJ7LNW** [comment]
This is the fix:

```diff
diff --git a/lib/PDL/IO/Touchstone.pm b/lib/PDL/IO/Touchstone.pm
index 7466fa6..c75ad21 100644
--- a/lib/PDL/IO/Touchstone.pm
+++ b/lib/PDL/IO/Touchstone.pm
@@ -146,7 +146,7 @@ sub rsnp_fh
        my $EOF_REGEX = $args->{EOF_REGEX};
 
        # Try to enforce the number of ports based on the filename extension.
-       $n_ports = $1 if ($fn =~ /s(\d+)p/i);
+       $n_ports = $1 if ($fn =~ /\.s(\d+)p$/i);
```

**Comment by mohawk2** [comment]
If it were "just a carp not a croak", then the test wouldn't have failed and I wouldn't have opened the issue.

**Comment by mohawk2** [comment]
> This is the fix:
> 
> ```diff
> diff --git a/lib/PDL/IO/Touchstone.pm b/lib/PDL/IO/Touchstone.pm
> index 7466fa6..c75ad21 100644
> --- a/lib/PDL/IO/Touchstone.pm
> +++ b/lib/PDL/IO/Touchstone.pm
> @@ -146,7 +146,7 @@ sub rsnp_fh
>         my $EOF_REGEX = $args->{EOF_REGEX};
>  
>         # Try to enforce the number of ports based on the filename extension.
> -       $n_ports = $1 if ($fn =~ /s(\d+)p/i);
> +       $n_ports = $1 if ($fn =~ /\.s(\d+)p$/i);
> ```

Great! Are you going to incorporate it, and release it, so we don't get spurious fails on PDL's CI?

**Comment by KJ7LNW** [comment]
30,000 tests, no problems.  I'll push a commit and do a release sometime soon.

**Comment by KJ7LNW** [comment]
Released 1.015
